src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -313,114 +385,141 @@ export async function updateUBAClient(
+                l1TokenInfo.symbol
+              );
+
+              // Construct the bundle data for this token.
+              const constructedBundle: UBABundleState = {
+                flows: [],
+                openingBlockNumberForSpokeChain: startingBundleBlockNumber,
+                openingBalance: runningBalance,
+                openingIncentiveBalance: incentiveBalance,
+                config: ubaConfigForBundle,
+              };
+
+              // TODO: Return a promise for each loop iteration and promise.all them
+              await Promise.all(
+                recentFlows.map(async (flow) => {
+                  // Previous flows will be populated with all flows we've stored into the `constructedBundle.flows`
+                  // array so far. On the first `flow`, this will be an empty array.
+                  const previousFlows = constructedBundle.flows.map((flow) => flow.flow);
+                  const previousFlowsIncludingCurrent = previousFlows.concat(flow);
+                  const {
+                    runningBalance: lastRunningBalance,
+                    incentiveBalance: lastIncentiveBalance,
+                    netRunningBalanceAdjustment,
+                  } = analog.calculateHistoricalRunningBalance(
+                    previousFlowsIncludingCurrent,
+                    constructedBundle.openingBalance,
+                    constructedBundle.openingIncentiveBalance,
+                    chainId,
+                    tokenSymbol,
+                    constructedBundle.config
+                  );
+                  const { balancingFee: depositBalancingFee } = getDepositFee(
+                    flow.amount,
+                    lastRunningBalance,
+                    lastIncentiveBalance,
+                    chainId,
+                    constructedBundle.config
+                  );
+                  const { balancingFee: relayerBalancingFee } = getRefundFee(
+                    flow.amount,
+                    lastRunningBalance,
+                    lastIncentiveBalance,
+                    chainId,
+                    constructedBundle.config
+                  );
+                  const lpFee = await computeLpFeeForRefresh(
+                    l1TokenAddress,
+                    cumulativeSpokeTargets,
+                    // @dev Assume that flow is taking refund on destination chain ID for purposes of computing the LP fee.
+                    // This is encoded in the UMIP as the way to compute utilization.
+                    flow.originChainId,
+                    flow.destinationChainId,
+                    flow.amount,
+                    hubPoolClient,
+                    constructedBundle.config.getBaselineFee(flow.destinationChainId, flow.originChainId),
+                    constructedBundle.config.getLpGammaFunctionTuples(flow.destinationChainId),
+                    flow.quoteBlockNumber
+                  );
```

Comment:
I'm unclear on why we compute the deposit fee, refund fee and LP fee for each flow. Is it just for simplicity, such that some of them will resolve to 0 because that type of fee is not relevant for the specific flow event?